docs/source/relayer.rst

```diff
@@ -0,0 +1,105 @@
+Cross-chain message relayer
+===========================
+
+| This component is responsible for handling the cross-chain contract state updates. 
+|
+| It is mainly used by the Agent software when resolving requests as a result of a running challenge game
+  or for sending fill invalidation messages (L1 resolutions). However, it can also be used manually by running
+  the CLI commands.
+| If you haven't read about the role of :ref:`l1-resolutions` in Beamer, make sure you get familiar 
+| with this concept first.
+|
+| Basically, every **request fill** action creates a **proof** on the request destination chain.
+  The agent utilizes these proofs in an optimistic way. If a challenge game ever occurs, or if a
+  fill was invalid, the agent has the chance to prove that a misbehavior occured and in the case of a challenge game,
+  get rewarded for it.  
+|
+| Since these **proofs** lay on a different chain than the chain where the requests originated from and where the challenge
+  game is running, we need a way to transport this **proof** from one chain to another in order to be able 
+  to prove that a certain state is correct. Thats where the cross-chain relayer comes into play.
+|
+| For instructions on how to setup, compile and run the relayer, make sure you take a look at the `/relayer/README.md` file.
+
```

Comment:
Why is `|` at the beginning of some lines above?

Could you please remove that and also the trailing whitespace in this file?

Reply:
I used it so i can preserve the line-breaks on some of the sentences.
For example, i wanted the sentence: "If you haven't read about the role ...." to stay in the same paragraph block but in a new line. When this paragraph block is not prefixed with "|", this sentence will continue on the same line as the previous sentence.

If you have a suggestion on how to avoid using "|" and achieve the desired thing, please let me know.

Reply:
> For example, i wanted the sentence: "If you haven't read about the role ...." to stay in the same paragraph block but in a new line. 

The question is, why would we want to do that?
Why not just start a new paragraph?

Reply:
I feel like that sentence should be kept with the others in that paragraph due to the context but if its on the same line the text becomes really hard to read. I noticed that in general with our whole documentation.

Reply:
But, i will remove it so we keep the same structure everywhere :)